insights/specs/datasources/db2.py

```diff
@@ -0,0 +1,64 @@
+"""
+Custom datasources to get the db2 users and database names.
+"""
+
+from insights.core.context import HostContext
+from insights.core.exceptions import SkipComponent
+from insights.core.plugins import datasource
+from insights.combiners.ps import Ps
+from insights.core.spec_factory import foreach_execute
+from insights.specs import Specs
+
+
+@datasource(Ps, HostContext)
+def db2_users(broker):
+    """
+    This datasource get the name of the users running db2 databases
+
+    Returns:
+        list: the names of the users running db2 databases
+
+    Raises:
+        SkipComponent: there is no db2 users running db2 databases
+    """
+    users_result = []
+    ps = broker[Ps]
+    ps_list = ps.search(COMMAND_NAME__contains="db2sysc")
+    if not ps_list:
+        raise SkipComponent("No db2 database is running")
+    for item in ps_list:
+        user_name = item.get("USER", None)
+        if user_name:
+            users_result.append(user_name)
+    if users_result:
+        users_result.sort()
+        return users_result
+    raise SkipComponent("No db2 user is available")
```

Comment:
1. Better remove the possible duplicated users from the result, e.g. by using a `set`
2. nitpick, the following code would be a bit simpler.

```python
if ps_list:
    users_result = set(filter(None, [item.get("USER") for item in ps_list]))
    if users_result:
        return sorted(users_result)
    raise SkipComponent("No db2 user is available")
raise SkipComponent("No db2 database is running")
```

Reply:
@xiangce - updated, please check